dockstore/dockstore-ui2
# Prevent snapshotting if no author set.

**Description**
Force the author(s) to be set before snapshotting, so that we can set the author field when creating the DOI. The author field is required when creating a DOI; we currently get around that by adding "Author unknown", which is not ideal; this will prevent that going forward.

The initial idea was to prevent publishing a DOI if the author isn't set, but since you can't add an author after you snapshot, I moved the check up earlier.

***Notes***
* We currently don't prevent this in the web service; that can be handled that in another PR. It looks like there's other enforcement the UI does that is not done by the web service; the ability to snapshot branches is only blocked in the UI.
* There are existing snapshots that slipped through without an author; there's not much we can do about that, since the versions are immutable.

***Alternative***
At the DB level, we don't freeze the authors for a version. It's not currently supported by the API, but in theory we could provide some way to add authors after snapshotting. Or maybe we should be freezing the authors for a frozen version.


**Review Instructions**
1. Try snapshotting a workflow version that does not have the author set; you should be unable to, and you should get the red box that is in the attached screenshot.
3. Try snapshotting a workflow version that does have the author set, you should be able to.

**Issue**
dockstore/dockstore#5437
https://ucsc-cgl.atlassian.net/browse/DOCK-2358


**Screenshot**
This comes up if the authors are not set after you click on the Snapshot menu choice.

![Screen Shot 2023-04-07 at 4 40 04 PM](https://user-images.githubusercontent.com/1049340/230693591-0b82d2d1-6e02-4dbe-aaf3-c70740696bff.png)


**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!
